// [[Reason: TSC detects forward ref which is not allowed and so enum value is undefined, which is also the way we detect isolated declaration errors.]] ////

//// [tests/cases/compiler/constEnumErrors.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -6,9 +6,9 @@
 }
 declare namespace E {
 }
 declare const enum E1 {
-    X = 0,
+    X,
     Y,
     Y1
 }
 declare const enum E2 {
@@ -35,8 +35,9 @@
 /// [Errors] ////
 
 constEnumErrors.ts(1,12): error TS2567: Enum declarations can only merge with namespace or other enum declarations.
 constEnumErrors.ts(5,8): error TS2567: Enum declarations can only merge with namespace or other enum declarations.
+constEnumErrors.ts(12,5): error TS9020: Enum member initializers must be computable without references to external symbols with --isolatedDeclarations
 constEnumErrors.ts(12,9): error TS2651: A member initializer in a enum declaration cannot reference members declared after it, including members defined in other enums.
 constEnumErrors.ts(14,5): error TS9020: Enum member initializers must be computable without references to external symbols with --isolatedDeclarations
 constEnumErrors.ts(14,9): error TS2474: const enum member initializers must be constant expressions.
 constEnumErrors.ts(14,12): error TS2339: Property 'Z' does not exist on type 'typeof E1'.
@@ -58,9 +59,9 @@
 constEnumErrors.ts(42,9): error TS2477: 'const' enum member initializer was evaluated to a non-finite value.
 constEnumErrors.ts(43,9): error TS2478: 'const' enum member initializer was evaluated to disallowed value 'NaN'.
 
 
-==== constEnumErrors.ts (23 errors) ====
+==== constEnumErrors.ts (24 errors) ====
     const enum E {
                ~
 !!! error TS2567: Enum declarations can only merge with namespace or other enum declarations.
         A
@@ -75,8 +76,10 @@
     const enum E1 {
         // illegal case
         // forward reference to the element of the same enum
         X = Y,
+        ~
+!!! error TS9020: Enum member initializers must be computable without references to external symbols with --isolatedDeclarations
             ~
 !!! error TS2651: A member initializer in a enum declaration cannot reference members declared after it, including members defined in other enums.
         // forward reference to the element of the same enum
         Y = E1.Z,
